examples/site/hello_world_pubsub/hello_world_pubsub.cc

```diff
@@ -45,10 +45,16 @@ std::string decode_base64(std::string const& base64) {
                            kBase64EncodedBits, kBase64RawBits>;
   // Pad the raw string if needed.
   auto padded = base64;
-  padded.append((4 - base64.size() % 4) % 4, '=');
-  auto const last_non_pad = base64.find_last_not_of('=');
-  auto const pad_count =
-      std::distance(padded.begin() + last_non_pad + 1, padded.end());
+  padded.append((4 - padded.size() % 4) % 4, '=');
+  auto pad_begin = [&padded] {
+    auto const last_non_pad = padded.find_last_not_of('=');
+    if (last_non_pad == std::string::npos) return padded.end();
+    auto r = padded.begin();
+    std::advance(r, last_non_pad + 1);
+    return r;
+  }();
+  auto const pad_count = std::distance(pad_begin, padded.end());
```

Comment:
I don't think I understand the complexity here.  Is it not equivalent to just:
```
auto const pad_count = (4 - padded.size() % 4) % 4;
padded.append(pad_count, '=');
```
[That even seems to better handle the should-not-happen case of a `base64` with a multiple-of-4 length that contains a `=`.]

Reply:
> I don't think I understand the complexity here. Is it not equivalent to just:

Sadly no, consider a string that should have two `=` and the end, but for whatever reason it has only one.  There is the case of `pad_count` in your formula yielding `3`, but that should not happen, as you pointed out.

> [That even seems to better handle the should-not-happen case of a `base64` with a multiple-of-4 length that contains a =.]

Which prompted me to change the code to handle any excessive padding.